lib/utils/report.mjs

```diff
@@ -129,6 +111,42 @@ export default function report(problem) {
 	result.warn(warningMessage, warningProperties);
 }
 
+/**
+ * @param {import('stylelint').Problem} problem
+ * @return {problem is import('stylelint').ProblemWithNormalizedPositions}
+ */
+function normalizeReportSourceLocation(problem) {
+	// Use the given node to calculate the start and end positions.
+	if (problem.node && (!problem.start || !problem.end)) {
+		const range = nodeRangeBy(problem.node, { index: problem.index, endIndex: problem.endIndex });
+
+		if (range) {
+			problem.start ??= range.start;
+			problem.end ??= range.end;
+		}
+	}
+
+	if (!isNumber(problem.line) && isNumber(problem.start?.line)) {
+		// Fallback the line to the line of the start position
+		problem.line = problem.start.line;
+	} else if (!problem.start && isNumber(problem.line)) {
+		// Fallback the start position to the line
+		problem.start = {
+			line: problem.line,
+			column: 1,
+		};
```

Comment:
I didn't add a fallback for the end position.
While I previously added this fallback:

```
warningProperties.end = {
 	line: startLine,
 	column: 2, // the start of the line + 1 when `node` is undefined
};
```

I am really unsure if we need a fallback for the end position and what that should be.

Reply:
I don't know if we should have a fallback.
But if we do, you should use `word`—if present and doesn't have `\n`—to deduce the end column.

Reply:
I don't think we should use `word` as a fallback for the lack of `end`.

`word` already has highest "priority": https://github.com/postcss/postcss/blob/1cc6ac3bf43a12460d3396e2cb76f14075daea1b/lib/node.js#L227-L237

If `word` is provided `start` and `end` won't be used at all by PostCSS.
So we don't have to use it as the fallback, it will be the primary value used by 
PostCSS.

This however is different from the logic in Stylelint and this could lead to differences between where an error is reported (the value calculated by PostCSS) and the location used to calculate disable ranges (the value calculated by Stylelint)

This difference is not new, it existed before these PR's.
If we want to make changes to the priority we should discuss so separately from this PR.

----------

I think the scope of the question for this PR is:

`end` already receives a fallback that will be used when `word` is not provided : https://github.com/postcss/postcss/blob/1cc6ac3bf43a12460d3396e2cb76f14075daea1b/lib/node.js#L217-L225

Given that we don't use `end` ourselves, do we then need to set a fallback to make it explicit?
Or should we leave it blank and leave it up to PostCSS?

Reply:
> Given that we don't use end ourselves, do we then need to set a fallback to make it explicit?
> Or should we leave it blank and leave it up to PostCSS?

At first I wanted to port the fallback to be in line with PostCSS but that wouldn't be a refactor PR anymore.
Since your stance is to limit the scope, I would say let's not add a fallback in this PR.
I would even say, if you can, remove more code.

Reply:
Since you 👍 Ill resolve this thread.